test/units/containers_service_test.rb

```diff
@@ -24,4 +24,15 @@ class ContainersServiceTest < ActiveSupport::TestCase
     Fog.unmock!
     assert_equal DockerContainerWizardState.where(:id => @state.id).count, 0
   end
+
+  test 'passes errors from compute resource' do
+    # Since the compute resource will be unreachable, this test will always
+    # fail at the 'pull_image' step
+    containers_service = Service::Containers.new
+    assert_raises(ActiveRecord::Rollback) do
+      containers_service.create_container_object(@state)
+    end
+    assert containers_service.errors.present?
```

Comment:
well, the array bug mentioned earlier would have been found if you had at least one test discussing the value of the errors array. i don't think we care too much about the actual errors (as they are generated by docker itself) I would only care about errors we raise internally and probably generic handling of errors coming from external sources?